packages/teams-js/src/public/profile.ts

```diff
@@ -0,0 +1,127 @@
+import { sendAndHandleSdkError as sendAndHandleError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { validateShowProfileRequest } from '../internal/profileUtil';
+import { FrameContexts } from './constants';
+import { ErrorCode } from './interfaces';
+import { runtime } from './runtime';
+
+export namespace profile {
+  /**
+   * Opens a profile card at a specified position to show profile information about a persona.
+   * @param showProfileRequest The parameters to position the card and identify the target user.
+   * @returns Promise that will be fulfilled when the operation has completed
+   */
+  export function showProfile(showProfileRequest: ShowProfileRequest): Promise<void> {
+    ensureInitialized(FrameContexts.content);
+
+    return new Promise<void>(resolve => {
+      if (!validateShowProfileRequest(showProfileRequest)) {
+        throw { errorCode: ErrorCode.INVALID_ARGUMENTS };
+      }
+
+      // Passing a DOMRect as part of the request breaks in Electron WebView with
+      // "Uncaught Error: An object could not be cloned.". To work around
+      // this we create a duplicate object of the values from the DOMRect.
+      const newShowProfileRequest = {
+        ...showProfileRequest,
+        targetElementBoundingRect: {
```

Comment:
Passing objects that contain functions out of this sdk has proven problematic in the past. You're working around one of the problems here by essentially making a new version that doesn't contain any functions, but we've seen that cause confusion in the past because another engineer will try to use the object and go "uh oh there are no functions what happened?". Might be better to just take the properties you need out of DomRect and put it into a sort of property bag. Will avoid confusion in the future.

Reply:
Fixed by explicitly declaring an internal type that the hub can take a dependency on, should make it. clear to anyone on the hub side what values we have available.